components/script/dom/gpuadapter.rs

```diff
@@ -60,4 +71,47 @@ impl GPUAdapterMethods for GPUAdapter {
     fn Extensions(&self, _cx: SafeJSContext) -> NonNull<JSObject> {
         NonNull::new(self.extensions.get()).unwrap()
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpuadapter-requestdevice
+    fn RequestDevice(&self, descriptor: &GPUDeviceDescriptor, comp: InCompartment) -> Rc<Promise> {
+        let promise = Promise::new_in_current_compartment(&self.global(), comp);
+        let sender = response_async(&promise, self);
+        let id = self.global().as_window().Navigator().create_device_id();
+        let desc = wgpu::DeviceDescriptor {
+            extensions: wgpu::Extensions {
+                anisotropic_filtering: descriptor.extensions.anisotropicFiltering,
+            },
+            limits: wgpu::Limits {
+                max_bind_groups: descriptor.limits.maxBindGroups,
+            },
+        };
+        match self.global().as_window().webgpu_channel() {
+            Some(thread) => thread
+                .0
+                .send(WebGPURequest::RequestDevice(sender, self.adapter, desc, id))
+                .unwrap(),
+            None => promise.reject_error(Error::Type("No WebGPU thread...".to_owned())),
+        }
+        promise
+    }
+}
+
+impl AsyncWGPUListener for GPUAdapter {
+    fn handle_response(&self, response: WebGPUResponse, promise: &Rc<Promise>) {
+        match response {
+            WebGPUResponse::RequestDevice(device_id, _descriptor) => {
+                let device = GPUDevice::new(
+                    &self.global(),
+                    &self,
+                    Heap::default(),
+                    Heap::default(),
```

Comment:
Will we create non-default values based on descriptor in the future?

Reply:
Yes, we will deal with it in a follow up patch.